tests/test_sequences_and_iterators.cpp

```diff
@@ -308,19 +336,63 @@ TEST_SUBMODULE(sequences_and_iterators, m) {
         .def("nonzero_keys", [](const IntPairs& s) {
             return py::make_key_iterator(NonZeroIterator<std::pair<int, int>>(s.begin()), NonZeroSentinel());
         }, py::keep_alive<0, 1>())
+        .def("nonzero_values", [](const IntPairs& s) {
+            return py::make_value_iterator(NonZeroIterator<std::pair<int, int>>(s.begin()), NonZeroSentinel());
+        }, py::keep_alive<0, 1>())
+        .def("simple_iterator", [](IntPairs& self) {
+            return py::make_iterator(self);
+        }, py::keep_alive<0, 1>())
+        .def("simple_keys", [](IntPairs& self) {
+            return py::make_key_iterator(self);
+        }, py::keep_alive<0, 1>())
+
+        // test iterator with keep_alive (doesn't work so not used at runtime, but tests compile)
```

Comment:
Wasn't this whole PR triggered by someone whose previously working code used py::keep_alive here?

Reply:
Yes, in https://github.com/pybind/pybind11/commit/ee0c5ee405e7a532410797687da28a20b89cd62b#r56899065. But they had an object that supported keep_alive, and here we don't, we just have a pair. I think you have to have a python wrapped iterator to use a second keep_live like this. This at least makes sure the keep_alive don't trigger the wrong overload - it will fail to compile if it does. If there's another handy Extra we could add, that would work too.

I'd be happy to add a keep_alive test, but don't want to write the proper class & iterator. Maybe @wojdyr could?

Reply:
Since we're just checking that SFINAE works rather than the behaviour of the policy, you could probably replace it with something like `py::call_guard<int>()` (which won't do anything useful).

Reply:
I've used that, and also added one for each of the other two.